### ISSUE TITLE
Broken testsuite

### ISSUE THREAD

**Issue by nijel** [python-babel/babel#106]
The Babel testsuite (and maybe the code as well, I did not check) is broken as it uses pytz API which is documented not to work. In short pytz timezones should not be used as tzinfo parameter to date/datetime functions, but rather the localize method should be used. See https://bugs.launchpad.net/pytz/+bug/1324158 for pytz bugreports and note in pytz documentation: http://pythonhosted.org//pytz/#introduction

```
[   42s] =================================== FAILURES ===================================
[   42s] ___________________ DateTimeFormatTestCase.test_timezone_gmt ___________________
[   42s] 
[   42s] self = <tests.test_dates.DateTimeFormatTestCase testMethod=test_timezone_gmt>
[   42s] 
[   42s]     def test_timezone_gmt(self):
[   42s]         tz = timezone('Europe/Berlin')
[   42s]         t = time(15, 30, tzinfo=tz)
[   42s]         fmt = dates.DateTimeFormat(t, locale='de_DE')
[   42s] >       self.assertEqual('GMT+01:00', fmt['ZZZZ'])
[   42s] E       AssertionError: 'GMT+01:00' != u'GMT+00:53'
[   42s] 
[   42s] tests/test_dates.py:188: AssertionError
[   42s] _________________ DateTimeFormatTestCase.test_timezone_rfc822 __________________
[   42s] 
[   42s] self = <tests.test_dates.DateTimeFormatTestCase testMethod=test_timezone_rfc822>
[   42s] 
[   42s]     def test_timezone_rfc822(self):
[   42s]         tz = timezone('Europe/Berlin')
[   42s]         t = time(15, 30, tzinfo=tz)
[   42s]         fmt = dates.DateTimeFormat(t, locale='de_DE')
[   42s] >       self.assertEqual('+0100', fmt['Z'])
[   42s] E       AssertionError: '+0100' != u'+0053'
[   42s] 
[   42s] tests/test_dates.py:182: AssertionError
[   42s] ____________________________ test_get_timezone_gmt _____________________________
[   42s] 
[   42s]     def test_get_timezone_gmt():
[   42s]         dt = datetime(2007, 4, 1, 15, 30)
[   42s]         assert dates.get_timezone_gmt(dt, locale='en') == u'GMT+00:00'
[   42s]     
[   42s]         tz = timezone('America/Los_Angeles')
[   42s]         dt = datetime(2007, 4, 1, 15, 30, tzinfo=tz)
[   42s] >       assert dates.get_timezone_gmt(dt, locale='en') == u'GMT-08:00'
[   42s] E       assert 'GMT-08:07' == 'GMT-08:00'
[   42s] E         - GMT-08:07
[   42s] E         ?         ^
[   42s] E         + GMT-08:00
[   42s] E         ?         ^
[   42s] 
[   42s] tests/test_dates.py:384: AssertionError
[   42s] ===================== 3 failed, 320 passed in 1.98 seconds =====================
```

**Comment by benselme** [comment]
Dupe of #133, fixed by #134

**Comment by etanol** [comment]
Already fixed, a couple of times apparently.  Closing.
